weblate/vcs/git.py

```diff
@@ -615,7 +590,39 @@ def push(self):
 
     def configure_remote(self, pull_url, push_url, branch):
         # We don't use push URL at all
-        super(GithubRepository, self).configure_remote(pull_url, None, branch)
+        super(_GitRemoteRepositoryMixin, self).configure_remote(
+            pull_url, None, branch)
+
+
+class GithubRepository(_GitRemoteRepositoryMixin):
+
+    name = 'GitHub'
+    _cmd = 'hub'
+    _username = settings.GITHUB_USERNAME
```

Comment:
This assigns a value to local variable at class initialization, what makes it impossible to override setting in the tests. It might be better to assign setting name here and use getattr to fetch actual value later.

Apparently running tests for GitHub pull requests was broken for some time, I've just fixed it in bbaadb62c696442c380b7c888173047ee7bc7e2b.

Reply:
Ah, that makes sense.

How about I convert `_username` to a `@property` instead?  It is slightly smaller and cleaner than multiple `getattr(settings, self._username_setting_name)`.

Reply:
Good idea.